src/prefect/engine/results/azure_result.py

```diff
@@ -0,0 +1,168 @@
+            - **kwargs (optional): if provided, will be used to format the location template
+                to determine the location to write to
+
+        Returns:
+            - Result: a new Result instance with the appropriately formatted location
+        """
+        new = self.format(**kwargs)
+        new.value = value
+
+        self.logger.debug("Starting to upload result to {}...".format(new.location))
+
+        ## prepare data
+        binary_data = new.serialize_to_bytes(new.value).decode()
+
+        # initialize client and upload
+        client = self.service.get_blob_client(
+            container=self.container, blob=new.location
+        )
+        client.upload_blob(binary_data)
+
+        self.logger.debug("Finished uploading result to {}.".format(new.location))
+
+        return new
+
+    def read(self, location: str) -> Result:
+        """
+        Reads a result from an Azure Blob container and returns a corresponding `Result` instance.
+
+        Args:
+            - location (str): the Azure blob location to read from
+
+        Returns:
+            - Result: the read result
+        """
+        new = self.copy()
+        new.location = location
+
+        try:
+            self.logger.debug("Starting to download result from {}...".format(location))
+
+            # initialize client and download
+            client = self.service.get_blob_client(
+                container=self.container, blob=location
+            )
+            content_string = client.download_blob()
+
+            try:
+                new.value = new.deserialize_from_bytes(content_string)
+            except EOFError:
+                new.value = None
+            self.logger.debug("Finished downloading result from {}.".format(location))
+        except Exception as exc:
+            self.logger.exception(
+                "Unexpected error while reading from result handler: {}".format(
+                    repr(exc)
+                )
+            )
+            raise exc
+        return new
+
+    def exists(self, location: str) -> bool:
+        """
+        Checks whether the target result exists.
+
+        Does not validate whether the result is `valid`, only that it is present.
+
+        Args:
+            - location (str): Location of the result in the specific result target.
+                Will check whether the provided location exists
+
+        Returns:
+            - bool: whether or not the target result exists.
+        """
+        from azure.core.exceptions import HttpResponseError
+
+        # initialize client and download
+        client = self.service.get_blob_client(container=self.container, blob=location)
+
+        # Catch exception because Azure python bindings do not yet have an exists method
+        # https://github.com/Azure/azure-sdk-for-python/issues/9507
+        try:
+            client.get_blob_properties()
+            return True
+        except HttpResponseError:
```

Comment:
From the issue in the comment linked above, it recommended catching a `ResourceNotFoundError` specifically for the exists problem (which according to the docs at least is for 404s, not sure it actually gets raised or not which maybe is why you picked this one: https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/azure/core/exceptions.py#L308), are we preferring `HttpResponseError` on purpose?